nmrfx-processor/src/main/java/org/nmrfx/processor/processing/Processor.java

```diff
@@ -1460,7 +1462,20 @@ public void closeDataset() {
         if (dataset != null) {
             if (dataset.isMemoryFile()) {
                 try {
-                    dataset.saveMemoryFile();
+                    if (dataset.getFileName().endsWith("data.dat") && (getNMRData() instanceof RS2DData)) {
```

Comment:
This is strange to have a RS2D specific save here. This will cause two problems:
1. it will be saved on error from `Processor.run(ProcessOps..)`, bottom of method.
2. this prevents a global notification on dataset save. 

I don't know the code base well enough to see what a better location would be. `DatasetBase.writeParFile(..)` could work for the header, but I don't know where you save the data for other file formats.

Reply:
I've simplified the code in closeDataset, but the RS2D specific save has to be somewhere.  Not sure where else to put it.  I looked into putting this code inside dataset.saveMemory file.  But I don't know if that is better than here, and it needs to have access to the rs2Ddata objet to save header info). 

The call from run(ProcessOps..) should probably not cause the call to write out datasets from memoryFile.  I'll check that and make sure it can be removed without a problem. That will eliminate problem 1

Reply:
That's what I was wondering: I didn't see a suitable central place. How/Where are other formats saved?